src/promptflow/promptflow/azure/_utils/_tracing.py

```diff
@@ -0,0 +1,85 @@
+# ---------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# ---------------------------------------------------------
+
+import typing
+
+from azure.ai.ml import MLClient
+from azure.core.exceptions import ResourceNotFoundError
+from azure.identity import AzureCliCredential, DefaultAzureCredential
+
+from promptflow._constants import CosmosDBContainerName
+from promptflow._sdk._tracing import get_ws_tracing_base_url
+from promptflow._sdk._utils import extract_workspace_triad_from_trace_provider
+from promptflow.azure import PFClient
+from promptflow.azure._restclient.flow_service_caller import FlowRequestException
+from promptflow.exceptions import ErrorTarget, UserErrorException
+
+
+def _get_credential() -> typing.Union[AzureCliCredential, DefaultAzureCredential]:
+    try:
+        credential = AzureCliCredential()
+        credential.get_token("https://management.azure.com/.default")
+        return credential
+    except Exception:
+        return DefaultAzureCredential()
+
+
+def _create_trace_provider_value_user_error(
+    message: str, no_personal_data_message: typing.Optional[str] = None
+) -> UserErrorException:
+    return UserErrorException(
+        message=message,
+        target=ErrorTarget.CONTROL_PLANE_SDK,
+        no_personal_data_message=no_personal_data_message,
```

Comment:
what's this? no_personal_data_message

Reply:
i undertstand this field will be collected by our telemetry as there is no personal data here?

\+ @D-W- and @Stephen1993 for precise explanation, not sure if i'm using it incorrectly.

Reply:
seems i'm using memory from v2... just removed this.